### DISCUSSION TITLE
How to work with date type in prisma and postgresql

### DISCUSSION THREAD

**Discussion by AlbertMontolio** [prisma/prisma#14773] [Q&A]
I have a `nestjs` app using `prisma`, with following `postgresql` table:

```
    Table "public.Transaction"
    Column |       Type               |               
    ------ -+-----------------+
    id           | integer                  |
    amount | double precision  |
    date      | date                       |
```

In prisma, I defined the table as this:

```
    model Transaction {
      id                Int      @id @default(autoincrement())
      amount            Float   
      date              DateTime @db.Date
```

Since I am using `postgresql`, I need to pass a date with `'YYYY-MM-DD'` format.

I am trying to pass this format as a string:

```
    const createdTransaction = await this.prisma.transaction.create({
      data: {
        amount: 3333,
        date: '2022-11-22',
      },
    });
```

But I get following error, rightfully so:

`    Argument date: Got invalid value '2022-11-22' on prisma.createOneTransaction. Provided String, expected DateTime.
`
I know I have to pass a Date type, but using the javascript formatting, everything ends up in a string.

What value should I pass to date within the `nestjs` app? Because I have js date objects, and postgresql accepts a date object, but that looks like a string 'YYYY-MM-DD'

If I use `new Date(2022,11,22)`, it is accepted by `prisma`, but this is stored as `Thu Dec 22 2022 00:00:00 GMT+0100 (Central European Standard Time)`, a string, but not the format that `postgresql` expects.

Many people told me to tell to prisma that date should be type `string`, then works fine for `prisma` and `postgresql`, but then I lose capabilities of doing queries like: 

```
SELECT 
        date_trunc('month', date) as date_month,
```

where I group by month.

Could anyone clarify what is the right way to work with `Date` type of `postgresql` in `prisma`. I found this in the `prisma` documentation for `postgresql`:

https://www.prisma.io/dataguide/postgresql/date-types

but this just shows within `postgresql`, the use is when using prisma in a node environment, where we have to do:

`this.prisma.transaction.create
`

**Comment by ludralph** [comment]
Hey Albert :wave:

We have this GitHub Issue: [#4355](https://github.com/prisma/prisma/issues/4355), which discusses allowing only dates of  format YYYY-MM-DD
For now, you could use string datatype to store the date literal, or you could use this [middleware](https://github.com/prisma/prisma/issues/4355#issuecomment-1026047496) which converts the date before sending the response to the client.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
